test/e2e/storage/csi_mock_volume.go

```diff
@@ -768,6 +935,84 @@ func checkPodLogs(cs clientset.Interface, namespace, driverPodName, driverContai
 	return nil
 }
 
+func parseMockLogs(cs clientset.Interface, namespace, driverPodName, driverContainerName string) ([]mockCSICall, error) {
+	// Load logs of driver pod
+	log, err := e2epod.GetPodLogs(cs, namespace, driverPodName, driverContainerName)
+	if err != nil {
+		return nil, fmt.Errorf("could not load CSI driver logs: %s", err)
```

Comment:
This failed because "the server rejected our request for an unknown reason (get pods csi-mockplugin-0)" (https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/89041/pull-kubernetes-e2e-gce-storage-slow/1245284273221537792/).

Perhaps a retry loop would help?

Reply:
There is already a retry loop in the caller (via `wait.Poll`). Perhaps failures to retrieve log output should simply be treated here as "no output", i.e. return `nil, nil`?

Reply:
Maybe the pod is not Running yet. The new mock test calls `createPod()` and immediately after that it reads the pod logs. I added wait for PVC to get bound in between - the driver must be fully operational to provision a PV.

Reply:
Sorry about the failed build noise. Stupid typo...

Reply:
I don't get it. The mock driver provisioned a PV, yet it gets error from API server:

`
could not load CSI driver logs: the server rejected our request for an unknown reason (get pods csi-mockplugin-0)
`

Reply:
> There is already a retry loop in the caller (via wait.Poll). Perhaps failures to retrieve log output should simply be treated here as "no output", i.e. return nil, nil?

I've implemented that idea in https://github.com/kubernetes/kubernetes/pull/88114 (https://github.com/kubernetes/kubernetes/pull/88114/commits/3cf9ab134a3201880472c063e5b527b88f3b0ebb) and got all tests to pass. The log does indeed show that "the server rejected our request for an unknown reason" occurred, but tests succeeded after ignoring that error (https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/88114/pull-kubernetes-e2e-kind/1245706940810530817/build-log.txt).

Reply:
I added your last commit from #88114, hoping it helps.

I can't reproduce "the server rejected our request for an unknown reason" on any of my clusters.

Reply:
> I can't reproduce "the server rejected our request for an unknown reason" on any of my clusters.

Me neither. I think saw it once while working on some other test, but as far as I remember, that then turned out to be because I was I was asking for logs after the pod had just been deleted. Perhaps here we have the inverse, asking for a very recently started pod? Just wondering.